### PR TITLE
Fix Morello + CHERI-RISC-V {make,set,swap}context

### PR DESCRIPTION
**Pull request by jrtc27**

### CODE REVIEW

File: sys/arm64/arm64/exec_machdep.c [hunk 1]
```diff
@@ -762,7 +762,11 @@ set_mcontext(struct thread *td, mcontext_t *mcp)
 
 	tf->tf_sp = mcp->mc_capregs.cap_sp;
 	tf->tf_lr = mcp->mc_capregs.cap_lr;
-	tf->tf_elr = mcp->mc_capregs.cap_elr;
+	if (SV_PROC_FLAG(td->td_proc, SV_UNBOUND_PCC))
```
Comment (bsdjhb) [inline sys/arm64/arm64/exec_machdep.c:765]: Would it be worth pushing the handling for `SV_UNBOUND_PCC` into `trapframe_set_elr`?

File: bin/cheribsdtest/cheribsdtest_context.c [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	p = mmap(NULL, len, PROT_READ | PROT_WRITE, MAP_ANON, -1, 0);
+	CHERIBSDTEST_VERIFY2(p != MAP_FAILED, "failed to map new stack");
+	uctx->uc_stack.ss_sp = p;
+	uctx->uc_stack.ss_size = len;
+}
+
+static void
+setcontext_func(int arg1, int arg2)
+{
+	CHERIBSDTEST_VERIFY(arg1 == 0x43485249);
```
Comment (bsdjhb) [inline bin/cheribsdtest/cheribsdtest_context.c:56]: Maybe #define's for ARG1 and ARG2?

**Review by bsdjhb — APPROVED** [review]
The *context API is rather bleh.  I used the kyua tests several years ago to verify the API for purecap MIPS.  They seemed to have decent coverage as well.

File: sys/arm64/arm64/exec_machdep.c (hunk 1 above)
Comment (jrtc27) [reply sys/arm64/arm64/exec_machdep.c:765]: That would do awful things for nested faults in a purecap kernel, no?

Comment (bsdjhb) [reply sys/arm64/arm64/exec_machdep.c:765]: Oof, yes.  One could do gross things checking the VA for user vs kernel, but probably best not to.

**Review by bsdjhb — APPROVED** [review]

File: bin/cheribsdtest/cheribsdtest_context.c [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	CHERIBSDTEST_VERIFY(arg2 == SETCONTEXT_ARG2);
+	cheribsdtest_success();
+}
+
+CHERIBSDTEST(setcontext_basic, "Check that setcontext works",
+    /*
+     * Currently happens to pass for c18n, possibly because makecontext and
+     * setcontext calls are done in the same function?
+     */
+    .ct_flaky_reason = XFAIL_FLAKY_C18N_CONTEXT)
```
Comment (jrtc27) [inline bin/cheribsdtest/cheribsdtest_context.c:69]: I think this is the right thing to do to ensure that if it breaks CI keeps passing?

Comment (dpgao) [reply bin/cheribsdtest/cheribsdtest_context.c:69]: Yes. Doing the setcontext in another compartment will likely cause things to break.

**Review by brooksdavis — APPROVED** [review]

**Pull request merged**